mocha/mocha.d.ts

```diff
@@ -46,55 +42,29 @@ declare var xit: Mocha.ITestDefinition;
 declare var test: Mocha.ITestDefinition;
 declare var specify: Mocha.ITestDefinition;
 
-declare function before(action: () => void): void;
-
-declare function before(action: (done: MochaDone) => void): void;
-
-declare function before(description: string, action: () => void): void;
-
-declare function before(description: string, action: (done: MochaDone) => void): void;
-
-declare function setup(action: () => void): void;
-
-declare function setup(action: (done: MochaDone) => void): void;
-
-declare function after(action: () => void): void;
-
-declare function after(action: (done: MochaDone) => void): void;
-
-declare function after(description: string, action: () => void): void;
-
-declare function after(description: string, action: (done: MochaDone) => void): void;
-
-declare function teardown(action: () => void): void;
-
-declare function teardown(action: (done: MochaDone) => void): void;
-
-declare function beforeEach(action: () => void): void;
-
-declare function beforeEach(action: (done: MochaDone) => void): void;
-
-declare function beforeEach(description: string, action: () => void): void;
-
-declare function beforeEach(description: string, action: (done: MochaDone) => void): void;
-
-declare function suiteSetup(action: () => void): void;
-
-declare function suiteSetup(action: (done: MochaDone) => void): void;
-
-declare function afterEach(action: () => void): void;
-
-declare function afterEach(action: (done: MochaDone) => void): void;
-
-declare function afterEach(description: string, action: () => void): void;
-
-declare function afterEach(description: string, action: (done: MochaDone) => void): void;
+interface MochaDone {
+    (error?: any): any;
+}
 
-declare function suiteTeardown(action: () => void): void;
+interface ActionFunction {
+    (done?: MochaDone): any | PromiseLike<any>
```

Comment:
do not use optional arguments in call back. the caller will call with a value all the time. the callee can decide to ignore it or not. but making it optional does not really mean that.